src/page/ProductDetailPage/index.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {useLocation} from 'react-router-dom';
+
+import DetailItem from 'component/DetailItem';
+
+import {DetailItemPageWrapper} from 'page/ProductDetailPage/style';
+
+export default function ProductDetailPage() {
+  const {
+    state: {itemImgURL, itemName, itemPrice, id, disable},
+  } = useLocation();
```

Comment:
> This hook returns the current [location](https://reactrouter.com/docs/en/v6/utils/location) object. This can be useful if you'd like to perform some side effect whenever the current location changes.

useLocation의 경우 useEffect와 함께 자주 사용되며 url변경이 일어나면 어떤 일을 해주고자 할때 자주 사용된다고 하네요. 
현재 접근 페이지 location을 가져올때만 사용하는 줄 알았는데 사용법에 대해 알아가네요!

### 참고자료
https://reactrouter.com/docs/en/v6/hooks/use-location
https://www.kindacode.com/article/react-router-uselocation-hook-tutorial-and-examples/

Reply:
url로 props로 전달하는 방식을 알려줘서 사용해보았는데 리트님이 주신 코멘트를 생각해보니 올바르지 않은 방식이네요. API콜을 하는 방식으로 변경했습니다! [e177a2a](https://github.com/woowacourse/react-shopping-cart/pull/74/commits/e177a2ad00a40888dd7165ebf31696b425976bff)

Reply:
다시 생각해보니 두 가지 방법이 가능할 것 같네요. 
1. Redux에 있는 ProductList에서 값을 가져오는 방법
2. id를 기반으로 API요청을 하는 방법

두번째 방법은 실시간 정보를 가져올 수 있는 대신 API 요청이 늘어나서 어떤 방법이 더 적절한지 모르겠네요. 이정도 API요청은 하는게 맞다고 생각이 들어서 두번째 방법으로 진행했습니다! 리뷰어님의 생각은 어떠신가요??

Reply:
네, 제 생각도 도리님과 같습니다.  
개별 상품 정보 하나 가져오는 거니까 id 기반으로 API 요청해도 좋다고 생각합니다.